src/sentry/ratelimits/utils.py

```diff
@@ -137,6 +142,18 @@ def get_organization_id_from_token(token_id: int) -> int | None:
             "api_token_id": token_id,
         }
     )
+
+    if not installations:
+        # If we can't find the installation with the token, we need to check if
+        # the token exists in the SentryAppInstallationToken, which has as a
+        # foreign key to the relevant SentryAppInstallation.
+        installations = app_service.get_many(
+            filter={
+                "status": SentryAppInstallationStatus.INSTALLED,
+                "installation_token_api_token_id": token_id,
+            }
+        )
```

Comment:
RPC methods aren't cheap. Could this condition be pushed down into the existing service method? Could this fallback to `installation_token_api_token_id` be part of how the `api_token_id` filter works? Do we have scenarios where we only want to identify an application by one token type but not the other?

Reply:
Hmm I was considering combining the logic when checking `api_token_id`, which would look something like:
```Python
if "api_token_id" in filters:
    original_query = query.copy()
    query = query.filter(api_token_id=filters["api_token_id"])
    # We must evaluate the query here, because the api token id may not be set in SentryAppInstallation
    # if it's an internal integration, in which case we need to check SentryAppInstallationToken
    if not len(query):
        installations_with_token_id = (
            SentryAppInstallationToken.objects.select_related("sentry_app_installation")
            .filter(api_token_id=filters["api_token_id"])
            .values_list("sentry_app_installation", flat=True)
        )
    
        query = original_query.filter(id__in=installations_with_token_id)
```
I was hesitant because I think we'd have to evaluate the query directly in `apply_filters` to know if we need to search the extra table, and I wasn't sure if it was bad to not lazily load this.

I'm not certain if there are scenarios where we only want to search one table, but we could make this opt-in by having the service pass both `api_token_id` and `installation_token_api_token_id` (which I didn't do in the example)

Reply:
Yeah that is kind of awkward, as it adds a query to every app.get_many() call. Another approach could be to add a new parameter like you have currently, and use that parameter to generate a query that uses a subquery to find the app. 

```python
            if "api_installation_token_id" in filters:
                from django.db.models import Q, Subquery
                subquery = SentryAppInstallationToken.objects.filter(
                    api_token_id=filters["api_installation_token_id"],
                ).values("sentry_app_installation_id")
                query = query.filter(
                    Q(api_token_id=filters["api_installation_token_id"])
                    | Q(id__in=Subquery(subquery))
                )
```

I've not tested this thoroughly but I think it will generate a query that does what you're looking for.

Reply:
Pretty cool, that did exactly what I was looking for. I didn't know you could write subqueries like that :O